Source/CarthageKit/Git.swift

```diff
@@ -37,6 +36,52 @@ public struct FetchCache {
 			lastFetchTimes[url] = Date().timeIntervalSince1970
 		}
 	}
+
+    fileprivate static func repositoriesLastFetchTime() -> TimeInterval? {
+        guard let data = try? Constants.Dependency.repositoriesURL.extendedAttribute(for: "\(Constants.bundleIdentifier).repositories.last.fetch.time") else {
+            return nil
+        }
+        return data.withUnsafeBytes { $0.pointee as TimeInterval }
+    }
+
+    public static func updateRepositoriesLastFetchTime() {
+        var now = Date().timeIntervalSince1970
+        let data = Data(bytes: &now, count: MemoryLayout<TimeInterval>.size)
+        try? Constants.Dependency.repositoriesURL.setExtendedAttribute(for: "\(Constants.bundleIdentifier).repositories.last.fetch.time", data: data)
+    }
+}
+
+/// URL extension to support operations on extended attributes of a file
+fileprivate extension URL {
+
+    enum ExtendedAttributesError: Error {
+        case cannotReadAttribute
+        case cannotSetAttribute
+    }
+
+    func extendedAttribute(for name: String) throws -> Data  {
+        let data = try withUnsafeFileSystemRepresentation { path -> Data in
```

Comment:
Curious -- what are your thoughts on using this approach rather than using the methods provided by `NSFileManager`? I've never used either personally.